synapse/rest/client/v1/push_rule.py

```diff
@@ -162,6 +161,21 @@ def notify_user(self, user_id):
         self.notifier.on_new_event("push_rules_key", stream_id, users=[user_id])
 
     def set_rule_attr(self, user_id, spec, val):
+        if spec["attr"] not in ("enabled", "actions"):
+            # for the sake of potential future expansion, shouldn't report
+            # 404 in the case of an unknown request so check it corresponds to
+            # a known attribute first.
+            raise UnrecognizedRequestError()
+
+        namespaced_rule_id = _namespaced_rule_id_from_spec(spec)
+        rule_id = spec["rule_id"]
+        is_default_rule = rule_id.startswith(".")
+        if is_default_rule:
+            if namespaced_rule_id not in BASE_RULE_IDS:
+                raise SynapseError(
+                    404, "Unknown rule %r" % (namespaced_rule_id,), Codes.NOT_FOUND
```

Comment:
probably prefer %s here?
```suggestion
                    404, "Unknown rule %s" % (namespaced_rule_id,), Codes.NOT_FOUND
```

Reply:
Aha, OK, I blindly copied `%r` from somewhere else. For my own reference, `%r` does `repr(…)`.